stingray/crossspectrum.py

```diff
@@ -12,9 +13,148 @@
 from stingray.gti import cross_two_gtis, bin_intervals_from_gtis, check_gtis
 import copy
 
-__all__ = ["Crossspectrum", "AveragedCrossspectrum", "coherence", "time_lag"]
+__all__ = ["Crossspectrum", "AveragedCrossspectrum",
+           "coherence", "time_lag", "cospectra_pvalue"]
 
 
+def _averaged_cospectra_cdf(xcoord, n):
+    """
+    Function calculating the cumulative distribution function for
+    averaged cospectra, Equation 19 of Huppenkothen & Bachetti (2018).
+
+    Parameters
+    ----------
+    xcoord : float or iterable
+        The cospectral power for which to calculate the CDF.
+
+    n : int
+        The number of averaged cospectra
+
+    Returns
+    -------
+    cdf : float
+        The value of the CDF at `xcoord` for `n` averaged cospectra
+    """
+    if np.size(xcoord) == 1:
+        xcoord = [xcoord]
+
+    cdf = np.zeros_like(xcoord)
+
+    for i, x in enumerate(xcoord):
+        prefac_bottom1 = scipy.misc.factorial(n - 1)
+        # print("x: " + str(x))
+        for j in range(n):
+            # print("j: " + str(j))
+            prefac_top = scipy.misc.factorial(n - 1 + j)
+            prefac_bottom2 = scipy.misc.factorial(
+                n - 1 - j) * scipy.misc.factorial(j)
+            prefac_bottom3 = 2.0 ** (n + j)
+
+            prefac = prefac_top / (
+            prefac_bottom1 * prefac_bottom2 * prefac_bottom3)
+
+            # print("prefac: " + str(prefac))
+            gf = -j + n
+
+            # print("gamma_fac: " + str(gf))
+            first_fac = scipy.special.gamma(gf)
+            # print("first_fac: " + str(first_fac))
+            if x >= 0:
+                second_fac = scipy.special.gammaincc(gf, n * x) * first_fac
+                # print("second_fac: " + str(second_fac))
+                fac = 2.0 * first_fac - second_fac
+            else:
+                fac = scipy.special.gammaincc(gf, -n * x) * first_fac
+
+            cdf[i] += (prefac * fac)
+        if np.size(xcoord) == 1:
+            return cdf[i]
+        else:
+            continue
+    return cdf
+
+
+def cospectra_pvalue(power, nspec):
+    """
+    the assumption that there is no periodic oscillation in the data.
```

Comment:
This docstring looks weird, this looks like another version of what is written below?